packages/ag-charts-community/src/chart/chart.ts

```diff
@@ -730,79 +715,68 @@ export abstract class Chart extends Observable implements AgChartInstance {
-        return this._axes;
-    }
+            for (const axis of newValue) {
+                if (oldValue?.includes(axis)) continue;
+                axis.attachAxis(this.axisGroup, this.axisGridGroup);
+            }
 
-    protected _series: Series<any>[] = [];
-    set series(values: Series<any>[]) {
+            this.zoomManager.updateAxes(newValue);
+        },
+    })
+    axes: ChartAxis[] = [];
+
+    @ActionOnSet<Chart>({
+        changeValue(newValue, oldValue) {
+            this.onSeriesChange(newValue, oldValue);
+        },
+    })
+    series: Series<any>[] = [];
+
+    private onSeriesChange(newValue: Series<any>[], oldValue?: Series<any>[]) {
         this.removeAllSeries();
-        this.seriesLayerManager.setSeriesCount(values.length);
-        values.forEach((series) => this.addSeries(series));
-    }
-    get series(): Series<any>[] {
-        return this._series;
-    }
+        this.seriesLayerManager?.setSeriesCount(newValue.length);
+
+        for (const series of newValue) {
+            if (oldValue?.includes(series)) continue;
 
-    private addSeries(series: Series<any>): boolean {
-        if (!this.series.includes(series)) {
-            this.series.push(series);
             if (series.rootGroup.parent == null) {
                 this.seriesLayerManager.requestGroup(series);
             }
-            this.initSeries(series);
-            return true;
-        }
-        return false;
-    }
 
-    private initSeries(series: Series<any>) {
-        const chart = this;
-        series.chart = {
-            get mode() {
-                return chart.mode;
-            },
-            get seriesRect() {
-                return chart.seriesRect;
-            },
-            placeLabels() {
-                return chart.placeLabels();
-            },
-        };
-        this.addSeriesListeners(series);
-        series.addChartEventListeners();
+            const chart = this;
+            series.chart = {
+                get mode() {
+                    return chart.mode;
+                },
+                get seriesRect() {
+                    return chart.seriesRect;
+                },
+                placeLabels() {
+                    return chart.placeLabels();
+                },
+            };
+
+            this.addSeriesListeners(series);
+            series.addChartEventListeners();
+        }
     }
 
     protected removeAllSeries(): void {
-        this.series.forEach((series) => {
+        this.series?.forEach((series) => {
             series.removeEventListener('nodeClick', this.onSeriesNodeClick);
             series.removeEventListener('nodeDoubleClick', this.onSeriesNodeDoubleClick);
             series.destroy();
 
             series.chart = undefined;
         });
-        this._series = []; // using `_series` instead of `series` to prevent infinite recursion
-
-        // Reset animation state.
-        this.animationRect = undefined;
-        this.animationManager.reset();
```

Comment:
🚨 This was added last week to correctly handle Integrated Charts animation scenarios (specifically, reset initial load animation on series changes) - we can't delay this until `Chart.destroy()` because that doesn't always get invoked (we attempt to re-use the `Chart` instance whenever possible)

Reply:
So we want to reset the animation manager on both the destruction of a chart OR when replacing the series array?

Reply:
Resolved offline